src/main/java/application/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -1,28 +1,91 @@
 package application.onlinebookstore.service.impl;
 
+import application.onlinebookstore.dto.BookDto;
+import application.onlinebookstore.dto.BookSearchParametersDto;
+import application.onlinebookstore.dto.CreateBookRequestDto;
+import application.onlinebookstore.exception.EntityNotFoundException;
+import application.onlinebookstore.mapper.BookMapper;
 import application.onlinebookstore.model.Book;
-import application.onlinebookstore.repository.BookRepository;
+import application.onlinebookstore.repository.book.BookRepository;
+import application.onlinebookstore.repository.book.impl.BookSpecificationBuilder;
 import application.onlinebookstore.service.BookService;
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
 import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+    private final BookSpecificationBuilder bookSpecificationBuilder;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto bookDto) {
+        Book savedBook = bookRepository.save(bookMapper.toModel(bookDto));
+        return bookMapper.toDto(savedBook);
+    }
+
+    @Override
+    public BookDto getBookById(Long id) {
+        return bookMapper.toDto(bookRepository.findById(id)
+                .orElseThrow(
+                        () -> new EntityNotFoundException("Can't find book by id: " + id)
+                ));
+    }
+
+    @Override
+    public List<BookDto> getBookByTitle(String title) {
+        List<Book> allByTitleContainsIgnoreCase = bookRepository
+                .findAllByTitleContainsIgnoreCase(title);
+        if (allByTitleContainsIgnoreCase.isEmpty()) {

```

Comment:
if returning a list you don't need to throw an exception if it's empty, returning an empty list is OK